platform/features/hyperlink/test/HyperlinkControllerSpec.js

```diff
@@ -0,0 +1,28 @@
+define(
+    ["../src/HyperlinkController"],
+    function (HyperlinkController) {
+
+        describe("The controller for hyperlinks", function () {
+            var scope,
+                domainObject,
+                model
+                link="http://nasa.gov"
+            beforeEach(function () {
+              domainObject = jasmine.createSpyObj(
+                  "domainObject",
+                  ["getModel"]
+              );
+              model = jasmine.createSpyObj(
+                "getModel",
+                ["link"]
+              );
+              domainObject.getModel.link= link
+            });
+            it("opens a specific given link", function () {
+                expect(domainObject.getModel.link)
+                    .toEqual("http://nasa.gov");
+            });
+
```

Comment:
Test the methods you defined on the controller, e.g.  "it knows when it should open in new tab" and "it knows when it is a button".